### PR TITLE
Start adding the Joystick API, with stubs for the browser

### PR DESCRIPTION
**Pull request by samhocevar**

### CODE REVIEW

File: browser.go [hunk 1]
```diff
@@ -759,6 +759,66 @@ const (
 	MouseButtonMiddle = MouseButton3
 )
 
+type Joystick int
+
+const (
+	Joystick1  Joystick = 1
+	Joystick2  Joystick = 2
```
Comment (Bluebugs) [inline browser.go:766]: Do we need to be explicit on those value?

File: browser.go [hunk 2]
```diff
@@ -759,6 +759,66 @@ const (
...
+	Joystick16 Joystick = 16
+
+	JoystickLast = Joystick16
+)
+
+type GamepadAxis int
+
+const (
+        AxisLeftX        GamepadAxis = 1
+        AxisLeftY        GamepadAxis = 2
```
Comment (Bluebugs) [inline browser.go:789]: Same question here and for all the const () block.

File: browser.go (hunk 1 above)
Comment (samhocevar) [reply browser.go:766]: You’re right, `iota` will do fine. I’m adding a commit to the PR.

Comment (Bluebugs) [reply browser.go]: Way better, thanks!

**Review by Bluebugs — APPROVED** [review]

**Pull request merged**